File: nx-monorepo/apps/admin/src/app/services/organization.service.ts
import { Injectable, signal } from '@angular/core';
import { collection, getFirestore, onSnapshot, orderBy, query } from 'firebase/firestore';
import { getFirebaseApp, Organization } from '@amdash/auth';

// Unconstrained across every organization — only legal under
// firestore.rules for a super-admin, so only ever inject this from the
// super-admin-gated /organizations route (see app.routes.ts's
// superAdminGuard). Every other admin page stays scoped to the caller's own
// org via the Cloud Functions it calls, never a direct client query here.
@Injectable({ providedIn: 'root' })
export class OrganizationService {
  private readonly firestore = getFirestore(getFirebaseApp());

  readonly organizations = signal<Organization[]>([]);

  constructor() {
    const organizationsQuery = query(collection(this.firestore, 'organizations'), orderBy('name'));
    onSnapshot(organizationsQuery, (snapshot) => {
      this.organizations.set(
        snapshot.docs.map((docSnapshot) => ({ id: docSnapshot.id, ...(docSnapshot.data() as Omit<Organization, 'id'>) })),
      );
    });
  }
}

File: flutter/apps/admin/lib/services/organization_service.dart
import 'package:amdash_core/amdash_core.dart';
import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';

/// Mirrors `apps/admin/src/app/services/organization.service.ts`: an
/// unconstrained live listener across *every* organization — only legal
/// under `firestore.rules` for a super-admin, and only ever injected from
/// the super-admin-gated `/organizations` route. Every other admin page
/// stays scoped to the caller's own org via the Cloud Functions it calls,
/// never a direct client query here.
final organizationsProvider = StreamProvider<List<Organization>>((ref) {
  return FirebaseFirestore.instance
      .collection('organizations')
      .orderBy('name')
      .snapshots()
      .map((snapshot) => snapshot.docs.map((doc) => Organization.fromFirestore(doc.id, doc.data())).toList());
});

// ownOrganizationProvider (this screen's own retention/CMEK/audit-logging/
// FHIR-export toggles all bind to it) moved to amdash_core's
// own_organization_service.dart once EMS and physician also needed to
// read fhirExportEnabled — still available here via the amdash_core
// import above, just no longer defined in this file.
